processor/src/host/advice/map.rs

```diff
@@ -60,3 +63,43 @@ impl Extend<(RpoDigest, Vec<Felt>)> for AdviceMap {
         self.0.extend(iter)
     }
 }
+
+impl Serializable for AdviceMap {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        target.write_u32(self.0.len() as u32);
+        for (key, values) in self.0.iter() {
+            key.write_into(target);
+            values.write_into(target);
+        }
```

Comment:
It may be possible to simplify this as `target.write((key, value))` - i.e., we should be able to write tuples directly.